File: kobciye/lib/core/widgets/app_logo.dart
import 'package:flutter/material.dart';
import '../constants/app_colors.dart';
import '../constants/app_text.dart';

/// Path to the bundled Kobciye wordmark artwork (added via `assets/logo/`).
const String _logoAssetPath = 'assets/logo/kobciye-logo.png';

/// Kobciye wordmark.
///
/// Prefers the bundled logo artwork ([_logoAssetPath]) when it can be loaded
/// and the surface is light enough for its dark-blue/gold colorway to read
/// well. Falls back to a drawn wordmark — "Kobciye" with the brand's
/// signature gold dot over the "i" — when the asset is missing (or when
/// [light] is true and we need a white-on-dark version instead).
///
/// Set [light] to true when placed on a dark/gradient background.
class AppLogo extends StatelessWidget {
  final double size;
  final bool light;
  final bool showTagline;

  const AppLogo({
    super.key,
    this.size = 40,
    this.light = false,
    this.showTagline = false,
  });

  @override
  Widget build(BuildContext context) {
    final wordmark = light
        ? _DrawnWordmark(size: size, color: Colors.white)
        : Image.asset(
            _logoAssetPath,
            height: size * 1.05,
            fit: BoxFit.contain,
            errorBuilder: (context, error, stackTrace) =>
                _DrawnWordmark(size: size, color: AppColors.primary),
          );

    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      mainAxisSize: MainAxisSize.min,
      children: [
        wordmark,
        if (showTagline) ...[
          SizedBox(height: size * 0.12),
          Text(
            'Learn • Grow • Succeed',
            style: AppText.caption.copyWith(
              color: light ? Colors.white.withOpacity(0.75) : AppColors.muted,
            ),
          ),
        ],
      ],
    );
  }
}

/// Text-drawn fallback wordmark — "Kobciye" with a gold dot over the "i" —
/// used when the logo artwork can't be loaded, or when a white wordmark is
/// needed on dark/gradient surfaces.
class _DrawnWordmark extends StatelessWidget {
  final double size;
  final Color color;

  const _DrawnWordmark({required this.size, required this.color});

  @override
  Widget build(BuildContext context) {
    final style = TextStyle(
      fontFamily: AppText.fontFamily,
      fontSize: size,
      fontWeight: FontWeight.w800,
      letterSpacing: -0.5,
      height: 1,
      color: color,
    );

    // "ı" (dotless i) lets us draw the wordmark's "i" without its dot, so we
    // can overlay our own gold dot exactly where the brand mark places it.
    final full = TextPainter(
      text: TextSpan(text: 'Kobcıye', style: style),
      textDirection: TextDirection.ltr,
    )..layout();
    final prefix = TextPainter(
      text: TextSpan(text: 'Kobc', style: style),
      textDirection: TextDirection.ltr,
    )..layout();
    final dotlessI = TextPainter(
      text: TextSpan(text: 'ı', style: style),
      textDirection: TextDirection.ltr,
    )..layout();

    final dotSize = size * 0.17;

    return SizedBox(
      width: full.width,
      height: full.height,
      child: Stack(
        clipBehavior: Clip.none,
        children: [
          Text('Kobcıye', style: style),
          Positioned(
            left: prefix.width + dotlessI.width / 2 - dotSize / 2,
            top: size * 0.04,
            child: Container(
              width: dotSize,
              height: dotSize,
              decoration: const BoxDecoration(color: AppColors.accent, shape: BoxShape.circle),
            ),
          ),
        ],
      ),
    );
  }
}

File: kobciye-rn/src/widgets/AppLogo.js
import React, { useState } from 'react';
import { View, Image, Text, StyleSheet } from 'react-native';
import { Colors } from '../constants/colors';

const LOGO_ASSET = require('../../assets/logo/kobciye-logo.png');

// Kobciye wordmark.
//
// Prefers the bundled logo artwork when it can be loaded and the surface
// is light enough for its dark-blue/gold colorway to read well. Falls back
// to a drawn wordmark — "Kobciye" with the brand's signature gold dot over
// the "i" — when the asset is missing, or when `light` is true (placed on
// a dark/gradient background, where the dark logo wouldn't read).
//
// `showTagline` is intentionally a no-op for the public Phase 1 build —
// the logo area shows only the "Kobciye" wordmark, no slogan underneath.
export default function AppLogo({ size = 40, light = false, showTagline = false }) {
  const [imageFailed, setImageFailed] = useState(false);

  const wordmark =
    light || imageFailed ? (
      <DrawnWordmark size={size} color={light ? Colors.white : Colors.primary} />
    ) : (
      <Image
        source={LOGO_ASSET}
        style={{ height: size * 1.05, width: size * 3.6 }}
        resizeMode="contain"
        onError={() => setImageFailed(true)}
      />
    );

  return (
    <View style={{ alignItems: 'flex-start' }}>
      {wordmark}
    </View>
  );
}

// Text-drawn fallback wordmark — "Kobciye" with a gold dot over the "i".
function DrawnWordmark({ size, color }) {
  const dotSize = size * 0.17;
  // Approximate horizontal position of the "i" within "Kobciye" at this
  // font size/weight — placed by eye to mirror the brand mark.
  const dotLeft = size * 2.34;

  return (
    <View style={{ width: size * 4.2, height: size * 1.25 }}>
      <Text style={[styles.wordmark, { fontSize: size, color, letterSpacing: -1.2 }]}>Kobciye</Text>
      <View
        style={[
          styles.dot,
          {
            width: dotSize,
            height: dotSize,
            borderRadius: dotSize / 2,
            left: dotLeft,
            top: size * 0.02,
          },
        ]}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  wordmark: { fontWeight: '800' },
  dot: { position: 'absolute', backgroundColor: Colors.accent },
});
